src/backend/distributed/planner/distributed_planner.c

```diff
@@ -1815,6 +1817,18 @@ multi_relation_restriction_hook(PlannerInfo *root, RelOptInfo *relOptInfo,
 	/* see comments on GetVarFromAssignedParam() */
 	relationRestriction->outerPlanParamsList = OuterPlanParamsList(root);
 
+	if (root->append_rel_list != NIL)
+	{
+		AppendRelInfo *targetAppendRelInfo = FindTargetAppendRelInfo(
+			root->append_rel_list, relationRestriction->index);
+		if (targetAppendRelInfo != NULL)
+		{
+			/* postgres deletes translated_vars after pg13, hence we copy them here */
+			relationRestriction->translatedVars = 
+				list_copy(targetAppendRelInfo->translated_vars);
```

Comment:
list_copy seems enough here because there are Vars in this list and we aren't interested in any pointers in the Vars.

Reply:
Well, lets be on the safe side an call `list_copy_deep`?

Reply:
list_copy_deep is only available in pg13 I think.

I will manually do the deep copy then.